zillow_address_pull.py

```diff
@@ -21,17 +23,99 @@ def random_address():
 
     return rent_zestimate, full_address, home_size, last_sold_price, last_sold_date
 
-# Pull housing data from the Zillow API
-# Note: Need to clean up a lot old/not used code
-def zillow_info(address, zipcode):
-    # import packages
-    import pyzillow
-    from pyzillow.pyzillow import ZillowWrapper, GetDeepSearchResults
-    import pandas as pd
 
+# Pull housing data from a user uploaded list of addresses. First it will check if the data already exists and if it does
+# then it will take the data from the database. If the data doesn't exist in the database, then it will pull it from
+# zillow.
+
+def zillow_info_mass_upload():
+    # upload dataset
+    housing_df = pd.read_excel('/home/bobby/zillow_project/housing_data.xlsx')
+    # upload addresses
+    addresses_df = pd.read_excel('/home/bobby/zillow_project/single_test_address.xlsx')
+
+    # make sure all addresses are capital for ease of mapping
+    addresses_df['address'] = addresses_df['address'].str.upper()
+
+    # merge the tables and see which ones are not in the table and which ones are
+    merged_df = addresses_df.merge(housing_df, on=['address', 'zipcode'], how='outer', indicator='exists')
+
+    # dataframe with info from internal database
+    matched_addresses = merged_df[merged_df['exists'] == 'both']
+    matched_addresses['full_address'] = matched_addresses['address'].astype(str) + matched_addresses['zipcode'].astype(
+        str)
+
+    # dataframe that needs to be pushed through Zillow's API
+    not_matched_addresses = merged_df[merged_df['exists'] == 'left_only']
+
+
+    if len(not_matched_addresses) > 0:
+        # run the non_matched addresses through Zillow's API
+        for i in range(len(not_matched_addresses)):
+
+            try:
```

Comment:
I'd throw the try/except into it's own function, and the logic to do 1 call in a function as well. That way this function can just handle the pandas logic and then have a very basic for loop that externs to a different function.